airflow/contrib/operators/gcp_translate_speech_operator.py

```diff
@@ -0,0 +1,156 @@
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from google.protobuf.json_format import MessageToDict
+
+
+from airflow import AirflowException
+from airflow.contrib.hooks.gcp_speech_to_text_hook import GCPSpeechToTextHook
+from airflow.contrib.hooks.gcp_translate_hook import CloudTranslateHook
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class GcpTranslateSpeechOperator(BaseOperator):
+    """
+    Recognizes speech in audio input and translates it.
+
+    Note that it uses the first result from the recognition api response - the one with the highest confidence
+    In order to see other possible results please use
+    :ref:`howto/operator:GcpSpeechToTextRecognizeSpeechOperator`
+    and
+    :ref:`howto/operator:CloudTranslateTextOperator`
+    separately
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:GcpTranslateSpeechOperator`
+
+    See https://cloud.google.com/translate/docs/translating-text
+
+    Execute method returns string object with the translation
+
+    This is a list of dictionaries queried value.
+    Dictionary typically contains three keys (though not
+    all will be present in all cases).
+
+    * ``detectedSourceLanguage``: The detected language (as an
+      ISO 639-1 language code) of the text.
+    * ``translatedText``: The translation of the text into the
+      target language.
+    * ``input``: The corresponding input value.
+    * ``model``: The model used to translate the text.
+
+    Dictionary is set as XCom return value.
+
+    :param audio: audio data to be recognized. See more:
+        https://googleapis.github.io/google-cloud-python/latest/speech/gapic/v1/types.html#google.cloud.speech_v1.types.RecognitionAudio
+    :type audio: dict or google.cloud.speech_v1.types.RecognitionAudio
+
+    :param config: information to the recognizer that specifies how to process the request. See more:
+        https://googleapis.github.io/google-cloud-python/latest/speech/gapic/v1/types.html#google.cloud.speech_v1.types.RecognitionConfig
+    :type config: dict or google.cloud.speech_v1.types.RecognitionConfig
+
+    :type target_language: str
+    :param target_language: The language to translate results into. This
```

Comment:
Do we have a list of strings allowed here?

Reply:
I've added link to the documentation with the list of available languages.